Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4992,27 +4995,29 @@ - (void)contactsPicker:(ContactsPicker *)contactsPicker didSelectContact:(Contac
 
     DDLogDebug(@"%@ in %s with contact: %@", self.logTag, __PRETTY_FUNCTION__, contact);
 
+    [self dismissViewControllerAnimated:YES completion:nil];
+
     OWSContact *_Nullable contactShareRecord = [OWSContacts contactForSystemContact:contact.cnContact];
     if (!contactShareRecord) {
         DDLogError(@"%@ Could not convert system contact.", self.logTag);
         return;
     }
 
     BOOL isProfileAvatar = NO;
-    UIImage *_Nullable avatarImage = contact.image;
+    NSData *_Nullable avatarImageData = contact.imageData;
```

Comment:
In order to get contact merging to work, I copped some ideas from your branch - we're now always passing around data to be sure we don't re-encode any images.